tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetry.Collector.cs

```diff
@@ -68,34 +66,57 @@ public void CopyTo(IConfigurationTelemetry destination)
-                    name: entry.Key,
-                    origin: entry.Origin.ToStringFast(),
-                    seqId: entry.SeqId,
-                    error: entry.Error,
-                    value: GetValue(entry));
+                destination.Add(GetConfigKeyValue(entry));
             }
+        }
 
-            static object? GetValue(ConfigurationTelemetryEntry entry)
+        private static object? GetValue(ConfigurationTelemetryEntry entry)
+        {
+            return entry.Type switch
             {
-                return entry.Type switch
-                {
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Bool => entry.BoolValue,
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Double => entry.DoubleValue,
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Int => entry.IntValue,
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Redacted => "<redacted>",
-                    _ => entry.StringValue
-                };
-            }
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Bool => entry.BoolValue,
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Double => entry.DoubleValue,
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Int => entry.IntValue,
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Redacted => "<redacted>",
+                _ => entry.StringValue
+            };
+        }
+
+        private static string? GetValueAsString(ConfigurationTelemetryEntry entry)
+        {
+            return entry.Type switch
+            {
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Bool => entry.BoolValue == null ? null : (entry.BoolValue.Value ? "true" : "false"),
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Double => entry.DoubleValue?.ToString("0.0###", CultureInfo.InvariantCulture),
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Int => entry.IntValue?.ToString(CultureInfo.InvariantCulture),
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Redacted => "<redacted>",
+                _ => entry.StringValue
+            };
+        }
+
+        private static ConfigurationKeyValue MapConfigKeyValueAsString(ConfigurationTelemetryEntry entry, string newKey)
+        {
+            var newEntry = ConfigurationTelemetryEntry.String(newKey, GetValueAsString(entry), entry.Origin, entry.Error);
+            return GetConfigKeyValue(newEntry);
+        }
+
+        private static ConfigurationKeyValue GetConfigKeyValue(ConfigurationTelemetryEntry entry)
+        {
+            return new ConfigurationKeyValue(
+                name: entry.Key,
+                origin: entry.Origin.ToStringFast(),
+                seqId: entry.SeqId,
+                error: entry.Error,
+                value: GetValue(entry));
```

Comment:
I don't think these are used anymore? 

```suggestion
```

Reply:
Right, removed in https://github.com/DataDog/dd-trace-dotnet/pull/5126/commits/46e0e94548c40d689bf5d8ce34e5962f86e350a3